frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkLoadManager.java

```diff
@@ -120,6 +131,29 @@ public int getNumRunningWorkers(String modelName) {
         }
     }
 
+    public void addServerThread(Model model, CompletableFuture<HttpResponseStatus> future)
+            throws InterruptedException, ExecutionException, TimeoutException {
+        WorkerStateListener listener = new WorkerStateListener(future, 1);
+        BatchAggregator aggregator = new BatchAggregator(model);
+        synchronized (model.getModelName()) {
+            model.setPort(port.getAndIncrement());
```

Comment:
This may be out of scope for this PR, but we should probably refactor this API to wrap the "getAndIncrement()". Something like ```model.getPortAndIncrement()``` or some such. What do you think?

Reply:
let's make this a separate pr if required